Categories/Category/Core.agda

```diff
@@ -108,8 +108,27 @@ record Category (o ℓ e : Level) : Set (suc (o ⊔ ℓ ⊔ e)) where
   CommutativeSquare : ∀ {A B C D} → (f : A ⇒ B) (g : A ⇒ C) (h : B ⇒ D) (i : C ⇒ D) → Set _
   CommutativeSquare f g h i = h ∘ f ≈ i ∘ g
 
+  {- Locally open equational reasoning module for clearer proofs. -}
+  module _ {A B : Obj} where
+    open HomReasoning {A} {B} public
+
   id-unique : ∀ {o} {f : o ⇒ o} → (∀ g → g ∘ f ≈ g) → f ≈ id
-  id-unique g∘f≈g = trans (sym identityˡ) (g∘f≈g id)
+  id-unique {o} {f} g∘f≈g = begin
+      f
+    ≈˘⟨ identityˡ ⟩
```

Comment:
I personally prefer making the combinator at the same line as the morphism given all mophisms are fairly short.